hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -227,6 +265,32 @@ public synchronized void reInitializeTasks(ReconOMMetadataManager omMetadataMana
     } catch (CancellationException ce) {
       LOG.error("Some tasks were cancelled with exception", ce);
     }
+
+    if (isRunSuccessful.get()) {
+      try {
+        reconDBProvider.replaceStagedDb(stagedReconDBProvider);
+        reconNamespaceSummaryManager.reinitialize(reconDBProvider);
+        reconContainerMetadataManager.reinitialize(reconDBProvider);
+        reprocessTaskStatus.setLastTaskRunStatus(0);
+        reprocessTaskStatus.recordRunCompletion();
```

Comment:
This need to be done carefully, because this is just switchover part and tasks already completed their reprocess logic above, so this whole operation/process should be either complete failed or complete success. In case of successful switchover only of STAGING DB to PROD DB, we can treat whole process as success and then only individual tasks status for reprocess should be updated as success else not. so need to refactor a code little bit for above `.thenAccept` code block.

Reply:
updated